app/controllers/sites/new.js

```diff
@@ -3,48 +3,117 @@ import { inject as service } from '@ember/service';
   }
 
+  async validateData() {
+    const { address, primaryContact, secondaryContact, site } = this.model;
+    const validationData = {
+      siteType: site.siteType.get('label'),
+      street: address.street,
+      country: address.country,
+      number: address.number,
+      postcode: address.postcode,
+      municipality: address.municipality,
+      province: address.province,
+      fullAddress: address.fullAddress,
+      telephonePrimary: primaryContact.telephone
+        ? primaryContact.telephone.replace(/ /g, '')
+        : '',
+      emailPrimary: primaryContact.email,
+      websitePrimary: primaryContact.website,
+      telephoneSecondary: secondaryContact.telephone,
+    };
+
+    const errorValidationResult = errorValidation.validate(validationData);
+    const warningValidationResult = warningValidation.validate(validationData);
+    return {
+      errors: errorValidationResult.error,
+      warnings: warningValidationResult.error,
+    };
+  }
+
   saveTask = task(async (event) => {
     event.preventDefault();
-    const { address, primaryContact, secondaryContact, site, adminUnit } =
-      this.model;
-    address.fullAddress = combineFullAddress(address);
-    await primaryContact.save();
-    await secondaryContact.save();
-    await address.save();
-    site.contacts = [primaryContact, secondaryContact];
-    site.address = address;
-    await site.save();
-    const nonPrimarySites = await adminUnit.sites;
-
-    if (this.isPrimarySite) {
-      const previousPrimarySite = await adminUnit.primarySite;
-      if (previousPrimarySite) {
-        nonPrimarySites.push(previousPrimarySite);
-      }
-      adminUnit.primarySite = site;
-    } else {
-      nonPrimarySites.push(site);
+    this.validationErrors = {};
+    this.validationWarnings = {};
+    const validationResult = await this.validateData();
+    if (
+      validationResult.errors &&
+      Object.keys(validationResult.errors).length > 0
+    ) {
+      // Handle errors
+      this.validationErrors = validationResult.errors.details.reduce(
+        (errors, detail) => {
+          errors[detail.context.key] = detail.message;
+          return errors;
+        },
+        {},
+      );
+      return;
+    }
+    if (
+      validationResult.warnings &&
+      validationResult.warnings.details.length > 0
+    ) {
+      console.log('er zijn warnings');
+      this.showWarningModal = true;
+      this.validationWarnings = validationResult.warnings.details.reduce(
+        (warnings, detail) => {
+          warnings[detail.context.key] = detail.message;
+          return warnings;
+        },
+        {},
+      );
+      return;
     }
+    // No errors and no warnings
+    await this.save();
+    this.reset();
+    this.router.transitionTo('sites.index');
+  });
 
-    await adminUnit.save();
-    this.router.transitionTo('sites.index'); // Model does not reload?
+  handleWarningModalOK = task(async (event) => {
```

Comment:
This is not supposed to be a task.
This should be an action which closes the dialog box and starts a save task.
In this way the user can see the loading animation.